client/src/document/hooks.ts

```diff
@@ -146,8 +146,10 @@ export function useFrequentlyViewed(): [
       localStorage.getItem(FREQUENTLY_VIEWED_STORAGE_KEY) || "[]"
     ) as FrequentlyViewedEntry[];
     const newEntries: FrequentlyViewedEntry[] = [];
+    let index = 0;
     for (const entry of entries) {
```

Comment:
changing this into a `for...in` loop and using the index as id should work. The lazy evaluation of `entry || index++` is dangerous. It should work but if it fails us it's tricky to debug.

Reply:
see https://github.com/mdn/yari/pull/6564#issuecomment-1211772056 for the explanation